linkchecker_test.go

```diff
@@ -0,0 +1,308 @@
+package linkchecker_test
+
+import (
+	"linkchecker"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestCrawl(t *testing.T) {
+
+	t.Parallel()
+
+	fs := http.FileServer(http.Dir("./testdata"))
+
+	ts := httptest.NewTLSServer(fs)
+
+	l, err := linkchecker.NewLinkChecker()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l.Domain = u.Host
+	l.Scheme = u.Scheme
+
+	l.HTTPClient = ts.Client()
+
+	url := ts.URL
+
+	want := []linkchecker.Result{
+		{
+			ResponseCode:  http.StatusOK,
+			Url:           ts.URL,
+			ReferringSite: ts.URL,
+		},
+		{
+			ResponseCode:  http.StatusOK,
+			Url:           ts.URL + "/about",
+			ReferringSite: ts.URL,
+		},
+		{
+			ResponseCode:  http.StatusOK,
+			Url:           ts.URL + "/home",
+			ReferringSite: ts.URL,
+		},
+		{
+			Url:           "mailto:home",
+			ReferringSite: ts.URL,
+		},
+	}
+
+	l.Crawl(url, url)
+
+	close(l.Results)
+
+	r := []linkchecker.Result{}
+
+	for result := range l.Results {
+
+		r = append(r, result)
+	}
+	got := r
+
+	if !cmp.Equal(want, got, cmpopts.IgnoreFields(linkchecker.Result{}, "Error")) {
+		t.Fatal(cmp.Diff(want, got))
+	}
+
+}
+
+func TestParseBody(t *testing.T) {
+	t.Parallel()
+
+	fs := http.FileServer(http.Dir("./testdata/about"))
+
+	ts := httptest.NewTLSServer(fs)
+
+	site := ts.URL
+
+	l, err := linkchecker.NewLinkChecker()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l.HTTPClient = ts.Client()
+
+	url, err := url.Parse(site)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l.Scheme, l.Domain = url.Scheme, url.Host
+
+	resp, err := l.HTTPClient.Get(site)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer resp.Body.Close()
+
+	want := []string{
+		ts.URL + "/home",
+		ts.URL + "/zzz",
+	}
+
+	got, err := l.ParseBody(resp.Body)
```

Comment:
If the test is for `ParseBody`, then I would _start_ it here. The extra paperwork only obscures the test, which seems to be asking "given some io.Reader, can I parse links out of it?"